src/_common/community/channel/channel-permissions.ts

```diff
@@ -0,0 +1,29 @@
+export class ChannelPermissions {
+	public static readonly ACTION_POSTING = 'posting';
+
+	allPermissions = false;
+	noPermissions = false;
+
+	perms?: { [key: string]: boolean };
+
+	constructor(input: any) {
+		if (!input) {
+			this.noPermissions = true;
+		} else if (input === true) {
+			this.allPermissions = true;
+		} else {
+			this.perms = input;
+		}
+	}
+
+	public canPerform(action: string) {
+		if (this.noPermissions) {
+			return false;
+		} else if (this.allPermissions) {
+			return true;
+		} else if (this.perms && this.perms[action] === true) {
```

Comment:
Prime candidate for safe navigation!
`} else if (this.perms?[action] === true) {`
I think that's the syntax when using [] at least.

Reply:
I don't think that's a thing, for direct array access you still need to check if the array is truthy.
`this.perms?[]` is not valid syntax.